src/dataExplorer/context/resultsView.tsx

```diff
@@ -47,19 +65,67 @@ export const ResultsViewProvider: FC = ({children}) => {
     } as SimpleTableViewProperties,
   })
 
-  const [viewOptions, setViewOptions] = useSessionStorage(
+  // what can be chosen (e.g. the list of all options)
+  const [viewOptionsAll, saveViewOptionsAll] = useSessionStorage(
+    'dataExplorer.resultsOptions.all',
+    DEFAULT_VIEW_OPTIONS
+  )
+  const setViewOptionsAll = useCallback(
+    (updatedOptions: Partial<ViewOptions>) => {
+      saveViewOptionsAll({...viewOptionsAll, ...updatedOptions})
```

Comment:
`setViewsOptionAll` and `saveViewOptionsAll` are really similarly named, and it takes a little bit of digging to understand the difference between the two. I can't think of any conventions around using set vs save - they're pretty interchangeable, so to avoid confusion, it might be worth calling this method `saveViewOptionsAllMemoized` to know that the only difference is this is memoized on `viewOptionsAll`

Reply:
I was going for `set` == "add to setting", and `save` == "save to local persistance". 🤷🏼